modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/IdentityProxy.java

```diff
@@ -0,0 +1,171 @@
+	private static IdentityProxy identityProxy = new IdentityProxy();
+	private static Context context;
+	public static String clientID;
+	public static String clientSecret;
+	private static String accessTokenURL;
+	private APIAccessCallBack apiAccessCallBack;
+	private TokenCallBack callback;
+	int requestCode = 0;
+
+	public int getRequestCode() {
+		return requestCode;
+	}
+
+	public void setRequestCode(int requestCode) {
+		this.requestCode = requestCode;
+	}
+
+	private IdentityProxy() {
+
+	}
+
+	public String getAccessTokenURL() {
+		return accessTokenURL;
+	}
+
+	public void setAccessTokenURL(String accessTokenURL) {
+		IdentityProxy.accessTokenURL = accessTokenURL;
+	}
+
+	@Override
+	public void receiveAccessToken(String status, String message, Token token) {
+		if (token != null) {
+			Log.d(TAG, token.getAccessToken());
+			Log.d(TAG, token.getRefreshToken());
+		}
+		IdentityProxy.token = token;
+		apiAccessCallBack.onAPIAccessRecive(status);
+	}
+
+	@Override
+	public void receiveNewAccessToken(String status, String message, Token token) {
+		IdentityProxy.token = token;
+		callback.onReceiveTokenResult(token, status);
+	}
+
+	public static synchronized IdentityProxy getInstance() {
+		return identityProxy;
+	}
+
+	/**
+	 * Initializing the IDP plugin and obrtaining the access token.
+	 * @param info - Includes token end point and Oauth app credentials.
+	 * @param apiAccessCallBack - Callback when API access happens.
+	 * @param appContext - Application context.
+	 */
+	public void init(CredentialInfo info, APIAccessCallBack apiAccessCallBack, Context appContext) {
+		IdentityProxy.clientID = info.getClientID();
+		IdentityProxy.clientSecret = info.getClientSecret();
+		this.apiAccessCallBack = apiAccessCallBack;
+		context = appContext;
+		SharedPreferences mainPref = context.getSharedPreferences(Constants.APPLICATION_PACKAGE
+		                                                          			, Context.MODE_PRIVATE);
+		Editor editor = mainPref.edit();
+		editor.putString(Constants.CLIENT_ID, clientID);
+		editor.putString(Constants.CLIENT_SECRET, clientSecret);
+		editor.putString(Constants.TOKEN_ENDPOINT, info.getTokenEndPoint());
+		editor.commit();
+		setAccessTokenURL(info.getTokenEndPoint());
+		AccessTokenHandler accessTokenHandler = new AccessTokenHandler(info, this);
+		accessTokenHandler.obtainAccessToken();
+	}
+
+	public void getToken(Context contextt, TokenCallBack call, String clientID, String clientSecret){
+		context = contextt;
+		callback = call;
+		IdentityProxy.clientID = clientID;
+		IdentityProxy.clientSecret = clientSecret;
+		if (token == null) {
+			getStoredToken();
+		} else {
+			boolean isExpired = ServerUtilities.isValid(token.getDate());
+			if (!isExpired) {
+				IdentityProxy.getInstance().receiveNewAccessToken(Constants.REQUEST_SUCCESSFUL, 
+				                                                  				"success", token);
+			} else {
+				getStoredToken();
+			}
+		}
+	}
+
+	private void getStoredToken(){
```

Comment:
Name of this method gives an impression that the method returns something but the signature is declared as "void". Please use some meaningful name that better explains what the method is supposed to do.